syft/execution/placeholder.py

```diff
@@ -24,7 +24,7 @@ def __init__(self, owner=None, id=None, tags: set = None, description: str = Non
         if not isinstance(self.id, PlaceholderId):
             self.id = PlaceholderId(self.id)
 
-        self.expected_shape = shape
+        self.expected_shape = tuple(shape) if shape is not None else None
```

Comment:
Starting to think we should maybe have a generic `Shape` class, even if it just inherits from `tuple`.

Reply:
(Not a change for this PR necessarily, just thinking out loud.)